src/pages/Loans/LoansOverview/components/LoanForm/LoanForm.tsx

```diff
@@ -139,10 +138,10 @@ const LoanForm = ({ asset, variant, position, overlappingModalRef, onChangeLoan
       case 'borrow':
         return transaction.execute(Transaction.LOANS_BORROW, monetaryAmount.currency, monetaryAmount);
       case 'repay': {
-        const isRepayAll = isMaxRepayAmount(monetaryAmount, assetAmount.max);
+        const isRepayAll = isMaxAmount(monetaryAmount);
```

Comment:
Brought the check for max amount into the hook where the max amounts are calculated